lib/src/merge.rs

```diff
@@ -415,6 +415,12 @@ impl<T> FromIterator<T> for MergeBuilder<T> {
     }
 }
 
+impl<T> FromIterator<T> for Merge<T> {
```

Comment:
iirc, `.collect()` into `Merge` is not implemented by design, and the caller should use `MergeBuilder` instead.

Reply:
Perhaps it would be better to add a `mut_map` method to Merge.

Reply:
I went with `into_map`. PTAL.

Reply:
I think I tried to explain the reason here:
https://github.com/martinvonz/jj/blob/eba9ac0b1b6cc8a564749598dd8101d60bceffe5/lib/src/merge.rs#L374-L379

However, I don't understand it myself now. Feel free to try to remove `MergeBuilder` and see how it goes. Maybe I was just confused (or maybe I'm missing something now).

Reply:
I also don't understand why `Option` matters, but I think the goal was to make `.collect()` not panic.